src/wallet/rpcdump.cpp

```diff
@@ -722,7 +726,8 @@ UniValue dumpwallet(const UniValue& params, bool fHelp)
         std::string strTime = EncodeDumpTime(it->first);
         std::string strAddr = CBitcoinAddress(keyid).ToString();
         CKey key;
-        pwalletMain->mapKeyMetadata[keyid].ParseComponents();
+        if(!masterKeyID.IsNull())
+            pwalletMain->mapKeyMetadata[keyid].ParseComponents();
```

Comment:
`ParseComponents()` function is crashing on invalid path (including empty path). This fixes non-HD wallet case but malformed path case remains. It needs to throw an exception instead